### PR TITLE
Handle internal auth

### PR DESCRIPTION
**Pull request by johnwatson484**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: e.g: Relates to ADO Work Item [AB#213700](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/213700) and builds on #3376 (link to ADO Build ID URL)*

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*

# Checklist (please delete before completing or setting auto-complete)
- [ ] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [ ] Title pattern should be `{work item number}: {title}`
- [ ] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests
- [ ] Version number in **package.json** has been updated

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: app/api/get-apim-headers.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    apimHeaders.crn = headers.crn
+  }
+
+  if (headers.email) {
+    apimHeaders.email = headers.email
+  }
+
+  if (headers.authorization) {
+    apimHeaders['X-Forwarded-Authorization'] = headers.authorization
+  }
```
Comment (johnwatson484) [inline app/api/get-apim-headers.js:19]: APIM request now only includes these headers if they were included on the original request

File: app/api/get-cache-key.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const getCacheKey = (path, headers) => {
+  return headers.email ? `${path}-internal` : `${path}-${headers.crn}`
```
Comment (johnwatson484) [inline app/api/get-cache-key.js:2]: If there is an email then the request is from an internal user so we cache the internal user view.  If it's not then it's from an external customer so we cache the view relevant to the external user.

File: app/api/get-host.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const { apimConfig } = require('../config')
+
+const getHost = (headers) => {
+  return headers.email ? apimConfig.hostInternal : apimConfig.hostExternal
```
Comment (johnwatson484) [inline app/api/get-host.js:4]: Thoughts are hosting the APIs again to resolve this issue so need to work out which host to use depending on if internal user

**Review by FayToward — APPROVED** [review]
LGTM

**Pull request merged**